geniza/corpus/forms.py

```diff
@@ -226,13 +226,15 @@ class DocumentSearchForm(forms.Form):
     }
 
     def filters_active(self):
+        # returns True if any filter fields are active in cleaned_data
         if self.is_valid():
-            return (
-                self.cleaned_data["doctype"]
-                or self.cleaned_data["has_transcription"]
-                or self.cleaned_data["has_translation"]
-                or self.cleaned_data["has_discussion"]
+            filter_fields = list(
+                filter(lambda k: k not in ["q", "sort"], self.fields.keys())
             )
+            active_fields = dict(
+                filter(lambda i: bool(i[1]), self.cleaned_data.items())
+            ).keys()
+            return bool(set(filter_fields).intersection(set(active_fields)))
```

Comment:
This looks more complicated than I think we need (and it's hard to read).

How about this?

```suggestion
            return bool(
                {
                    k: v
                    for k, v in self.cleaned_data.items()
                    if k not in ["q", "sort"] and bool(v)
                }
            )
```

Reply:
Looks much cleaner! Didn't realize we didn't need to actually compare it with `self.fields`, great solution.